zstash/hpss_utils.py

```diff
@@ -12,7 +12,35 @@
 import _io
 
 from .hpss import hpss_put
-from .settings import BLOCK_SIZE, TupleFilesRowNoId, config, logger
+from .settings import BLOCK_SIZE, TupleFilesRowNoId, TupleTarsRowNoId, config, logger
+from .utils import create_tars_table, tars_table_exists
+
+
+# Minimum output file object
+class HashIO(object):
+    def __init__(self, name: str, mode: str):
+        self.f: _io.BufferedWriter = open(name, mode)
+        self.hash: _hashlib.HASH = hashlib.md5()
```

Comment:
Perhaps we could have a `do_hash=True` option.
`self.do_hash = do_hash`

In `write()` and `md5()` update the hash *if* `do_hash` is `True`.